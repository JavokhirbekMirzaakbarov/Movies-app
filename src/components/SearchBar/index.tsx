import { Input } from "@mui/material";
import React from "react";
import style from "./styles.module.scss";

const SearchBar = () => {
  return (
    <Input
      className={style.searchBar}
      placeholder="What do you want to watch?"
      margin="dense"
      autoFocus
      disableUnderline
      fullWidth
    />
  );
};

export default SearchBar;
