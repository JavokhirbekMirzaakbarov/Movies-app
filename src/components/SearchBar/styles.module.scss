.searchBar {
  background-color: rgba(150, 150, 150, 0.4);
  height: 52px;
  border-radius: 2px;
  border: none;
  color: white;
  font-size: large;
  padding-left: 10px;

  &::placeholder {
    color: lightgray;
  }
}
