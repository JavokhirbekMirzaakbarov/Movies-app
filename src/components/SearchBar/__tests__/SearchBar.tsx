import React from "react";
import { screen, render } from "@testing-library/react";
import SearchBar from "../index";

describe("testing search bar", () => {
  test("renders correctly", () => {
    render(<SearchBar searchValue="some value" setSearchValue={() => ({})} />);
    expect(screen.getByDisplayValue("some value")).toBeInTheDocument();
  });

  it("does not change", () => {
    const { asFragment } = render(
      <SearchBar searchValue="some value" setSearchValue={() => ({})} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
