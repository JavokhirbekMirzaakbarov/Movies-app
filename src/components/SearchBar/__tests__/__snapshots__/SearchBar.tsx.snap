// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing search bar does not change 1`] = `
<DocumentFragment>
  <div
    class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused searchBar css-1s97tdc-MuiInputBase-root-MuiInput-root"
  >
    <input
      class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      placeholder="What do you want to watch?"
      type="text"
      value="some value"
    />
  </div>
</DocumentFragment>
`;
